db/roles.ts

```diff
@@ -0,0 +1,48 @@
+import { getConnection } from "./index";
+
+export async function createRolesTable() {
+  const client = await getConnection();
+
+  const {
+    rows: [{ exists: rolesExists }],
+  } = await client.query(`
+      SELECT EXISTS( SELECT 1 FROM pg_tables WHERE schemaname='public' and tablename='roles');
+    `);
+
+  if (!rolesExists) {
+    await client.query(
+      `
+        CREATE TABLE roles (
+          roleId uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+          roleName varchar (64) UNIQUE NOT NULL
+        )
+      `
+    );
+    return true;
+  }
+  return false;
+}
+
+export async function createUsersRolesTable() {
+  const client = await getConnection();
+
+  const {
+    rows: [{ exists: usersRolesExists }],
+  } = await client.query(`
+      SELECT EXISTS( SELECT 1 FROM pg_tables WHERE schemaname='public' and tablename='users_roles');
+    `);
+
+  if (!usersRolesExists) {
+    await client.query(
+      `
+        CREATE TABLE users_roles (
```

Comment:
Looks pretty good. Just one thing, I think we need to add a `UNIQUE` requirement for the field combination of `userId` and `roleId`. The `userRoleId` field probably is not needed. 

https://stackoverflow.com/questions/14221775/in-postgresql-force-unique-on-combination-of-two-columns

Reply:
Love that idea. I wasn't sure how to get the combination of the two other fields to be unique without making them each unique. This solution solves that and makes an easy Primary.

Requested change made and pushed.

Reply:
Looks great!